ansible_ai_connect/main/ssl_manager.py

```diff
@@ -0,0 +1,180 @@
+        self.combined_ca_bundle = os.environ.get("COMBINED_CA_BUNDLE_PATH")
+        self.service_ca_path = os.environ.get("SERVICE_CA_PATH")
+        self.requests_ca_bundle = os.environ.get("REQUESTS_CA_BUNDLE")
+
+        logger.info("SSL Manager: Infrastructure mode initialized")
+        logger.debug("SSL Manager: Combined CA bundle: %s", self.combined_ca_bundle)
+        logger.debug("SSL Manager: Service CA: %s", self.service_ca_path)
+        logger.debug("SSL Manager: Requests CA bundle: %s", self.requests_ca_bundle)
+
+    def _get_legacy_service_ca_path(self) -> Optional[str]:
+        """Get legacy service CA path from Django settings with lazy loading.
+        This is to ensure SSL manager can be imported safely in any context.
+        """
+        try:
+            from django.conf import settings
+
+            return getattr(settings, "SERVICE_CA_PATH", None)
+        except (ImportError, AttributeError, Exception):
+            # Django not configured, settings not available, or ImproperlyConfigured
+            return None
+
+    def _get_ca_bundle_path(self) -> Optional[str]:
+        """Get the best available CA bundle path with prioritized fallback.
+        Priority order:
+        1. Infrastructure-provided combined bundle (operator creates this)
+        2. REQUESTS_CA_BUNDLE environment variable
+        3. Service CA from environment
+        4. Legacy service CA from settings
+        5. None (use system defaults)
+        Returns:
+            Path to CA bundle file or None for system defaults
+        """
+        # Infrastructure-provided combined bundle
+        if self.combined_ca_bundle and os.path.exists(self.combined_ca_bundle):
+            logger.debug(
+                "SSL Manager: Using infrastructure combined CA bundle: %s", self.combined_ca_bundle
+            )
+            return self.combined_ca_bundle
+
+        # Environment REQUESTS_CA_BUNDLE
+        if self.requests_ca_bundle and os.path.exists(self.requests_ca_bundle):
+            logger.debug("SSL Manager: Using REQUESTS_CA_BUNDLE: %s", self.requests_ca_bundle)
+            return self.requests_ca_bundle
+
+        # Service CA (for internal services)
+        if self.service_ca_path and os.path.exists(self.service_ca_path):
+            logger.debug("SSL Manager: Using service CA: %s", self.service_ca_path)
+            return self.service_ca_path
+
+        # Legacy settings service CA (lazy loaded)
+        legacy_service_ca = self._get_legacy_service_ca_path()
+        if legacy_service_ca and os.path.exists(legacy_service_ca):
+            logger.debug("SSL Manager: Using legacy service CA: %s", legacy_service_ca)
+            return legacy_service_ca
+
+        # No custom CA bundle found - use system defaults
+        logger.debug("SSL Manager: No custom CA bundle found, using system defaults")
+        return None
+
+    def get_requests_session(self) -> requests.Session:
```

Comment:
We can actually prepare a session that would reuse `get_ssl_context()`. I've prepared an example here for verify=False:

https://goneri.lebouder.net/allow-broken-certificate-for-a-specific-host-with-pythons-requests/

We can easily reuse the logic to prepare another Adapter for the other cases.